chain-abstraction/sol-implementation/assertion_chain.go

```diff
@@ -216,6 +227,24 @@ func NewAssertionChain(
 		return nil, err
 	}
 	chain.specChallengeManager = specChallengeManager
+	if chain.fastConfirmSafeAddress != (common.Address{}) {
+		safe, err := contractsgen.NewSafe(chain.fastConfirmSafeAddress, chain.backend)
+		if err != nil {
+			return nil, err
+		}
+		chain.fastConfirmSafe = safe
+		owners, err := safe.GetOwners(chain.GetCallOptsWithDesiredRpcHeadBlockNumber(&bind.CallOpts{Context: ctx}))
+		if err != nil {
+			return nil, err
+		}
+		chain.fastConfirmSafeOwners = owners
+		threshold, err := safe.GetThreshold(chain.GetCallOptsWithDesiredRpcHeadBlockNumber(&bind.CallOpts{Context: ctx}))
+		if err != nil {
+			return nil, err
+		}
+		chain.fastConfirmSafeThreshold = threshold.Uint64()
+		chain.fastConfirmSafeApprovedHashesOwners = make(map[common.Hash]map[common.Address]bool)
```

Comment:
Maybe let's wrap the two methods with retry.UntilSucceed calls so that we don't die here due to a flakey RPC connection

Reply:
Done.